SpockOpaleye/src/Utils.hs

```diff
@@ -0,0 +1,9 @@
+module Utils where
+
+import AppM
+import Control.Exception
+import Exceptions
+
+returnOneIfNE :: (Monad m) => String -> [a] -> m a
+returnOneIfNE _ (x:_) = return x
+returnOneIfNE msg _ = throw $ NotFoundException msg
```

Comment:
is this not compiling if replaced with https://hackage.haskell.org/package/safe-exceptions-0.1.4.0/docs/Control-Exception-Safe.html#v:throw You might have to do define a `MonadThrow` instance for `AppM`, which might not be such a bad idea as a core building block, btw.